### ISSUE TITLE
[BUG] Bug on pressing ENTER in multi select questions

### ISSUE THREAD

**Issue by vikaspatil0021** [formbricks/formbricks#1395]
### Issue Summary

https://www.loom.com/share/55e0c3aee5e843ac88a78a2828679172
Here is a Video of the bug

1. When I select an option and press ENTER in multi select question, the option is deselected with an error.
![Screenshot 2023-10-22 113712](https://github.com/formbricks/formbricks/assets/112180774/18eb5434-da87-4d69-af6e-a25eee8c87ae)

2. when I select 2 or more options and press ENTER then it directs me to next question but when I click back button and go back I see that  the last selected option is deselected again 

### Steps to Reproduce

1. create a survey with multi select question
2. follow the video

### Expected behavior

1. After selecting an option, when I press ENTER it should direct me to next question.
2. After selecting 2 or more option it should not deselect the last selected option.

### Environment

- [ ] Formbricks Cloud (app.formbricks.com)
- [ ] Self-hosted Formbricks

### Desktop (please complete the following information)

```markdown
- OS: windows 10
- Node: 18.18.2
```

**Comment by vikaspatil0021** [comment]
@jobenjada Could you please consider adding these labels **formtribe-2023** and **hacktoberfest**.

**Comment by vikaspatil0021** [comment]
@jobenjada  

https://www.loom.com/share/84c62713f95b4364bbae8a0525f4326c

Hey the issue I am facing is something like this
1. when I select 1 or more options in multi select question
2. I pressed ENTER key on keyboard to move to next question instead of clicking on next button on the web page
3. After completing the survey when I check the responses recorded
3. In the responses the last selected option in multi select question is not recorded
    (ex: In the above video, "Palms" option was not recorded)
    
Note: I pressed ENTER key instead of normal click on "next" button

I believe that the code I removed in my PR above is responsible for deselecting of last selected option in multi select question
